railties/lib/rails/application/configuration.rb

```diff
@@ -170,20 +170,23 @@ def session_store(*args)
           case @session_store
           when :disabled
             nil
-          when :active_record_store
-            begin
-              ActionDispatch::Session::ActiveRecordStore
-            rescue NameError
-              raise "`ActiveRecord::SessionStore` is extracted out of Rails into a gem. " \
-                "Please add `activerecord-session_store` to your Gemfile to use it."
-            end
           when Symbol
             ActionDispatch::Session.const_get(@session_store.to_s.camelize)
           else
             @session_store
           end
         else
           @session_store = args.shift
+
+          if @session_store == :active_record_store
```

Comment:
> Does that benefit outweigh a deprecation cycle? Seems a bit to quick in my opinion.

Yup, it does. 

> I'm curious which tests were failing?

With my earlier implementation, the test https://github.com/rails/rails/blob/81b6bc99754e33f31d4ed61306680ae2c6f68045/railties/test/application/configuration_test.rb#L1181 was failing. With the current implementation, it no longer fails but I have kept the commit as it is. Because I feel the error about missing `activerecord-session_store` gem should be shown when someone tries to set `config.session_store` to `:activerecord_store` and not when somebody tries to access `config.session_store`.

Currently the same `session_store` method is used for accessing as well as setting the session store.

``` ruby
def session_store(*args)
        if args.empty?
          # Access session store
          case @session_store
          when :disabled
            nil
          when :active_record_store
            begin
              ActionDispatch::Session::ActiveRecordStore
            rescue NameError
              raise "`ActiveRecord::SessionStore` is extracted out of Rails into a gem. " \
                "Please add `activerecord-session_store` to your Gemfile to use it."
            end
          when Symbol
            ActionDispatch::Session.const_get(@session_store.to_s.camelize)
          else
            @session_store
          end
        else
          # Set session store
          @session_store = args.shift
          @session_options = args.shift || {}
        end
      end
```
This is the implementation from master branch which raises the error in the code path of `accessing session store`. This change moves the error from `accessing session store` to `setting session store` code path. The behavior is same. IMO it will be good if we raise as early as possible.